Tests/RxComposableArchitectureTests/TestStoreTests.swift

```diff
@@ -0,0 +1,442 @@
+    
+        internal func testExpectedStateEqualityMustModify() async {
+            struct State: Equatable {
+                var count: Int = 0
+            }
+            
+            enum Action: Equatable {
+                case noop, finished
+            }
+            
+            let reducer = Reduce<State, Action>({ state, action in
+                switch action {
+                case .noop:
+                    return Effect(value: .finished)
+                case .finished:
+                    return .none
+                }
+            })
+            
+            let store = TestStore(
+                initialState: State(),
+                reducer: reducer,
+                useNewScope: true
+            )
+            
+            _ = await store.send(.noop)
+            await store.receive(.finished)
+            
+            XCTExpectFailure {
+                _ = store.send(.noop) {
+                    $0.count = 0
+                }
+            }
+            
+            XCTExpectFailure {
+                store.receive(.finished) {
+                    $0.count = 0
+                }
+            }
+        }
+    #endif
+    
+    internal func testStateAccess() {
+        enum Action { case a, b, c, d }
+        
+        let store = TestStore(
+            initialState: 0,
+            reducer: Reduce<Int, Action>({ count, action in
+                switch action {
+                case .a:
+                    count += 1
+                    return .merge(.init(value: .b), .init(value: .c), .init(value: .d))
+                case .b, .c, .d:
+                    count += 1
+                    return .none
+                }
+            }),
+            useNewScope: false
+        )
+        
+        store.send(.a) {
+            $0 = 1
+            XCTAssertEqual(store.state, 0)
+        }
+        XCTAssertEqual(store.state, 1)
+        
+        store.receive(.b) {
+            $0 = 2
+            XCTAssertEqual(store.state, 1)
+        }
+        XCTAssertEqual(store.state, 2)
+        
+        store.receive(.c) {
+            $0 = 3
+            XCTAssertEqual(store.state, 2)
+        }
+        XCTAssertEqual(store.state, 3)
+        
+        store.receive(.d) {
+            $0 = 4
+            XCTAssertEqual(store.state, 3)
+        }
+        
+        XCTAssertEqual(store.state, 4)
+    }
+    
+    internal func testStateAccess_useNewScope() async {
+        enum Action { case a, b, c, d }
+        
+        let store = TestStore(
+            initialState: 0,
+            reducer: Reduce<Int, Action>({ count, action in
+                switch action {
+                case .a:
+                    count += 1
+                    return .merge(.init(value: .b), .init(value: .c), .init(value: .d))
+                case .b, .c, .d:
+                    count += 1
+                    return .none
+                }
+            }),
+            useNewScope: true
+        )
+        
```

Comment:
I think better to create separate file for oldScope dik, and for this file, use newScope

Reply:
sure, we can do that @jeffersonsetiawan on next commit ya